satpy/tests/test_scene.py

```diff
@@ -1796,6 +1796,32 @@ def test_resample_ancillary(self):
         new_scene = scene.resample(dst_area)
         assert new_scene['comp20'] is new_scene['comp19'].attrs['ancillary_variables'][0]
 
+    def test_resample_multi_ancillary(self):
+        """Test that multiple ancillary variables are retained after resampling.
+
+        This test corresponds to GH#2329
+        """
```

Comment:
Are there other tests for resampling with ancillary variables? Or maybe just the data walker? I'm wondering if an existing test could be updated with the checks you include at the bottom of this test that would serve the same purpose as this test.

Reply:
There is `test_resample_ancillary`, which tests the case with a single ancillary variable.  The bugs #2329 and #2330 occur only if there is more than one ancillary variable.  I don't understand the data walker, but I don't think the tests involving it do anything with ancillary variables.

Reply:
> I don't understand the data walker, but I don't think the tests involving it do anything with ancillary variables.

Well that's unfortunate then because that was the whole point of that function/generator iirc.

Reply:
There doesn't seem to be any test specifically for `dataset_walker`.  The only mention in the test suite is in a mock:

```
$ find . -name '*.py' -exec grep dataset_walker {} +
./test_scene.py:                mock.patch('satpy.dataset.dataset_walker') as ds_walker:
```